src/components/MultiSelect/index.js

```diff
@@ -0,0 +1,240 @@
+import React, { useState, useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Label from '../Input/label';
+import { useUniqueIdentifier, useOutsideClick, useErrorMessageId } from '../../libs/hooks';
+import {
+    StyledInput,
+    StyledContainer,
+    ChipContainer,
+    StyledButtonIcon,
+    StyledPlaceholder,
+} from './styled';
+import InternalDropdown from '../InternalDropdown';
+import InternalOverlay from '../InternalOverlay';
+import RenderIf from '../RenderIf';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import PlusIcon from './icons/plus';
+import { ENTER_KEY, SPACE_KEY } from '../../libs/constants';
+import { getChips, positionResolver } from './helpers';
+
+const MultiSelect = props => {
+    const {
+        className,
+        style,
+        label,
+        hideLabel,
+        placeholder,
+        error,
+        bottomHelpText,
+        required,
+        disabled,
+        readOnly,
+        tabIndex,
+        variant,
+        chipVariant,
+        value,
+        onChange,
+        onFocus,
+        onBlur,
+        children,
+    } = props;
+    const [isOpen, setIsOpen] = useState(false);
+
+    const inputRef = useRef();
+    const triggerRef = useRef();
+    const dropdownRef = useRef();
+
+    const inputId = useUniqueIdentifier('input');
+    const errorMessageId = useErrorMessageId();
+
+    useEffect(() => {
+        if (isOpen) {
+            dropdownRef.current.focus();
+        }
+    }, [isOpen]);
+
+    const handleChange = val => {
+        inputRef.current.focus();
+        setIsOpen(false);
+        return onChange(val);
+    };
+
+    const handleDelete = option => {
+        if (Array.isArray(value)) {
+            return onChange(value.filter(val => val !== option));
+        }
+        return onChange([]);
+    };
+
+    const handleTriggerClick = () => {
+        if (disabled || readOnly) return;
+        setIsOpen(!isOpen);
+        if (!isOpen) {
+            // eslint-disable-next-line no-use-before-define
+            startListeningOutsideClick();
+        }
+    };
+
+    const handleKeyDown = event => {
+        if (event.target !== inputRef.current) return;
+        if (event.keyCode === ENTER_KEY || event.keyCode === SPACE_KEY) {
+            event.preventDefault();
+            if (disabled || readOnly) return;
+            setIsOpen(!isOpen);
+            if (!isOpen) {
+                // eslint-disable-next-line no-use-before-define
+                startListeningOutsideClick();
+            }
+        }
```

Comment:
since this pattern is being used twice, I think we should create a handler for this like:
```
const toggleDropdown = () => {
 if (disabled || readOnly) return;
        setIsOpen(!isOpen);
        if (!isOpen) {
            // eslint-disable-next-line no-use-before-define
            startListeningOutsideClick();
        }
}
```